Author: 	gerald
Date: 	Thu Mar 20 19:18:33 2008 UTC (2 days, 16 hours ago)
Log Message: 	

Don't call cap_set_proc() unless we were started with elevated privileges.
Otherwise, we might print dumpcap: cap_set_proc() fail return: Operation not
permitted to stderr.

--- trunk/dumpcap.c	2008/03/20 00:30:47	24703
+++ trunk/dumpcap.c	2008/03/20 19:18:33	24704
@@ -508,12 +508,15 @@
 
     relinquish_special_privs_perm();
 
-    print_caps("Post drop, pre set");
-    cap_set_flag(caps, CAP_EFFECTIVE,   cl_len, cap_list, CAP_SET);
-    if (cap_set_proc(caps)) {
-        cmdarg_err("cap_set_proc() fail return: %s", strerror(errno));
+    if (started_with_special_privs()) {
+        print_caps("Post drop, pre set");
+        cap_set_flag(caps, CAP_EFFECTIVE,   cl_len, cap_list, CAP_SET);
+        if (cap_set_proc(caps)) {
+            cmdarg_err("cap_set_proc() fail return: %s", strerror(errno));
+        }
+        print_caps("Post drop, post set");
     }
-    print_caps("Post drop, post set");
+
     cap_free(caps);
 }
 #endif /* HAVE_LIBCAP */
